### DISCUSSION TITLE
[Parte 2] Combinación TRADE-PUFFLE BUY-PUFFLE, orden puffles

### DISCUSSION THREAD

**Discussion by MystEdu** [IIC2133-PUC/2024-2#23] [Tarea 0] [answered]
Hola buenas,
Me surgieron dudas respecto a la posición en que quedan los puffles.
Para dar contexto de mi duda, pongo el siguiente ejemplo:

Penguin_1 tiene puffles en posiciones {0, 1, 2, 3}
Penguin_2 tiene puffles en posiciones {0, 1, 2, 3, 4, 5}

Si ocurre TRADE-PUFFLE 1 2 0 5

¿Como quedarían las posiciones? ¿De la siguiente forma?

Penguin_1 {0, 1, 2, 3, 5}
Penguin_2 {0[P1] 0, 1, 2, 3, 4}

Luego, si ocurre BUY-PUFFLE 1 4 color, ¿dónde debería ir el puffle comprado?
- Penguin_1 {0, 1, 2, 3, 4, 5} manteniendo orden de posición por trade
- Penguin_1 {0, 1, 2, 3, 5, 4} perdiendo orden de posición, pero manteniendo por último agregado

**Comment by 4n1t4** [comment, marked as answer]
Un trade es básicamente un swap.
Por ejemplo: 

P1={a,b,c}
P2={g,d,o,p}
TRADE 1 2 0 3
P1={p,b,c}
P2={g,d,o,a}
BUY 1 f....
P1={p,b,c,f}

**Reply by MystEdu** [reply]
aaaa mala mia me confundí con las posiciones, ahora me di cuenta XD. Es verdad que es un swap, muchas gracias
